src/main/java/duke/model/Meal.java

```
@@ -0,0 +1,149 @@
+package duke.model;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+
+/**
+ * Task is a public class that represents the tasks in duke.
+ * A task object encapsulates the description of the task, the type of task it is, and whether
+ * the task is done.
+ * @author Chua Zong Wei
+ */
+public class Meal {
+    protected String description;
+    protected String type = "";
+    protected boolean isDone;
+    protected SimpleDateFormat dateparser = new SimpleDateFormat("dd/MM/yyyy");
+    protected String date = dateparser.format(Calendar.getInstance().getTime());
+    protected HashMap<String, Integer> nutritionValue = new HashMap<String, Integer>();
+
+    /**
+     * This is the constructor of Task object.
+     * @param description the description of the task
+     */
+    public Meal(String description, String details) {
+        this.description = description.trim();
+        //todo: date input can only be accepted at the back of the statement
+        if (details.contains("/date")) {
+            String[] splitString = details.split("/date", 2);
+            try {
+                Date day;
+                day = dateparser.parse(splitString[1].trim());
+                this.date = dateparser.format(day);
+            } catch (Exception e) {
+                //todo something here
```

yes! please do something here! and not have an empty catch block. 
such code in the production branch is almost criminal!